pkg/controllers/provisioning/provisioner.go

```diff
@@ -382,6 +385,9 @@ func (p *Provisioner) Create(ctx context.Context, n *scheduler.NodeClaim, opts .
 		return "", err
 	}
 	nodeClaim := n.ToNodeClaim()
+	if nodeClaim.Spec.TerminationGracePeriod == nil {
```

Comment:
nit: would it make more sense to handle this in `ToNodeClaim` rather than in `Create`? That seems more natural to me since it's where we insert other implicit fields (e.g. the owner ref).

Reply:
Yeah, that's fair I can shift it there